File: Samples/VB.Net/VBDotNet/ISmpReportSettingDlg.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports SalesMatePlusLib

Namespace CSharpSDKTest
	Partial Public Class ISmpReportSettingDlg
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub PrintShopNameAtHeaderbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles PrintShopNameAtHeaderbtn.Click
			Dim m_SmpReportSettings As ISmpReportSettings

			m_SmpReportSettings = New SmpReportSettingsClass()

			Dim lPrntShpNme As Long

			lPrntShpNme = m_SmpReportSettings.PrintShopNameAtHeader

			If lPrntShpNme = 1 Then
				PrntShpNmeAtHederlbl.Text = "Enabled"
			ElseIf lPrntShpNme = 0 Then
				PrntShpNmeAtHederlbl.Text = "Disabled"
			End If
		End Sub

		Private Sub PrintShopLogobtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles PrintShopLogobtn.Click
			Dim m_SmpReportSettings As ISmpReportSettings

			m_SmpReportSettings = New SmpReportSettingsClass()

			Dim lPrntShpLogo As Long

			lPrntShpLogo = m_SmpReportSettings.PrintShopLogo

			If lPrntShpLogo = 1 Then
				PrntShpLogolbl.Text = "Enabled"

			ElseIf lPrntShpLogo = 0 Then
				PrntShpLogolbl.Text = "Disabled"
			End If
		End Sub

		Private Sub PrintPhNoAtFooterbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles PrintPhNoAtFooterbtn.Click
			Dim m_SmpReportSettings As ISmpReportSettings

			m_SmpReportSettings = New SmpReportSettingsClass()

			Dim lPrntPhno As Long

			lPrntPhno = m_SmpReportSettings.PrintPhNoAtFooter

			If lPrntPhno = 1 Then
				PrntPhnolbl.Text = "Enabled"

			ElseIf lPrntPhno = 0 Then
				PrntPhnolbl.Text = "Disabled"
			End If

		End Sub

		Private Sub PrintTimeAtFooterbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles PrintTimeAtFooterbtn.Click
			Dim m_SmpReportSettings As ISmpReportSettings

			m_SmpReportSettings = New SmpReportSettingsClass()

			Dim lprntTme As Long

			lprntTme = m_SmpReportSettings.PrintTimeAtFooter

			If lprntTme = 1 Then
				PrntTmeAtFootrlbl.Text = "Enabled"

			ElseIf lprntTme = 0 Then
				PrntTmeAtFootrlbl.Text = "Disabled"
			End If

		End Sub

		Private Sub BypassPrintDlgbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles BypassPrintDlgbtn.Click
			Dim m_SmpReportSettings As ISmpReportSettings

			m_SmpReportSettings = New SmpReportSettingsClass()

			Dim lBypassprnt As Long

			lBypassprnt = m_SmpReportSettings.BypassPrintDlg

			If lBypassprnt = 1 Then
				BypassPrntlbl.Text = "Enabled"

			ElseIf lBypassprnt = 0 Then
				BypassPrntlbl.Text = "Disabled"
			End If
		End Sub

		Private Sub AllwaysPrintMembershipFormbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles AllwaysPrintMembershipFormbtn.Click
			Dim m_SmpReportSettings As ISmpReportSettings

			m_SmpReportSettings = New SmpReportSettingsClass()

			Dim lPrntMbershpFrm As Long

			lPrntMbershpFrm = m_SmpReportSettings.AllwaysPrintMembershipForm

			If lPrntMbershpFrm = 1 Then
				prntMbrShplbl.Text = "Enabled"

			ElseIf lPrntMbershpFrm = 0 Then
				prntMbrShplbl.Text = "Disabled"
			End If
		End Sub

		Private Sub ReportSettingDlg_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load

		End Sub
	End Class
End Namespace

File: Samples/C#.Net/CSharpSDKTest/ISmpReportSettingDlg.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using SalesMatePlusLib;

namespace CSharpSDKTest
{
    public partial class ISmpReportSettingDlg : Form
    {
        public ISmpReportSettingDlg()
        {
            InitializeComponent();
        }

        private void PrintShopNameAtHeaderbtn_Click(object sender, EventArgs e)
        {
            ISmpReportSettings m_SmpReportSettings;

            m_SmpReportSettings = new SmpReportSettingsClass();

            long lPrntShpNme;

            lPrntShpNme = m_SmpReportSettings.PrintShopNameAtHeader;

            if (lPrntShpNme == 1)
            {
                PrntShpNmeAtHederlbl.Text = "Enabled";
            }
            else if (lPrntShpNme == 0)
            {
                PrntShpNmeAtHederlbl.Text = "Disabled";
            }
        }

        private void PrintShopLogobtn_Click(object sender, EventArgs e)
        {
            ISmpReportSettings m_SmpReportSettings;

            m_SmpReportSettings = new SmpReportSettingsClass();

            long lPrntShpLogo;

            lPrntShpLogo = m_SmpReportSettings.PrintShopLogo;

            if (lPrntShpLogo == 1)
            {
                PrntShpLogolbl.Text = "Enabled";
            }

            else if (lPrntShpLogo == 0)
            {
                PrntShpLogolbl.Text = "Disabled";
            }
        }

        private void PrintPhNoAtFooterbtn_Click(object sender, EventArgs e)
        {
            ISmpReportSettings m_SmpReportSettings;

            m_SmpReportSettings = new SmpReportSettingsClass();

            long lPrntPhno;

            lPrntPhno = m_SmpReportSettings.PrintPhNoAtFooter;

            if (lPrntPhno == 1)
            {
                PrntPhnolbl.Text = "Enabled";
            }

            else if (lPrntPhno == 0)
            {
                PrntPhnolbl.Text = "Disabled";
            }

        }

        private void PrintTimeAtFooterbtn_Click(object sender, EventArgs e)
        {
            ISmpReportSettings m_SmpReportSettings;

            m_SmpReportSettings = new SmpReportSettingsClass();

            long lprntTme;

            lprntTme = m_SmpReportSettings.PrintTimeAtFooter;

            if (lprntTme == 1)
            {
                PrntTmeAtFootrlbl.Text = "Enabled";
            }

            else if (lprntTme == 0)
            {
                PrntTmeAtFootrlbl.Text = "Disabled";
            }

        }

        private void BypassPrintDlgbtn_Click(object sender, EventArgs e)
        {
            ISmpReportSettings m_SmpReportSettings;

            m_SmpReportSettings = new SmpReportSettingsClass();

            long lBypassprnt;

            lBypassprnt = m_SmpReportSettings.BypassPrintDlg;

            if (lBypassprnt == 1)
            {
                BypassPrntlbl.Text = "Enabled";
            }

            else if (lBypassprnt == 0)
            {
                BypassPrntlbl.Text = "Disabled";
            }
        }

        private void AllwaysPrintMembershipFormbtn_Click(object sender, EventArgs e)
        {
            ISmpReportSettings m_SmpReportSettings;

            m_SmpReportSettings = new SmpReportSettingsClass();

            long lPrntMbershpFrm;

            lPrntMbershpFrm = m_SmpReportSettings.AllwaysPrintMembershipForm;

            if (lPrntMbershpFrm == 1)
            {
                prntMbrShplbl.Text = "Enabled";
            }

            else if (lPrntMbershpFrm == 0)
            {
                prntMbrShplbl.Text = "Disabled";
            }
        }

        private void ReportSettingDlg_Load(object sender, EventArgs e)
        {

        }
    }
}